### PR TITLE
[ntuple] Add more test coverage for RNTuple+RVec

### PR DESCRIPTION
**Pull request by eguiraud**
This PR brings coverage of RNTuple+RVec almost on par with RNTuple+std::vector, e.g. by templating some existing std::vector tests over the collection type.

There is one issue: in `RNTuple.ComplexRVec`, the element type's constructor and destructor gets called a different number of times for RVec than for std::vector, but I _think_ RVec is right and std::vector is wrong because of https://github.com/root-project/root/issues/10520 . I'd like to hear what Jakob thinks though :)

### CODE REVIEW

**Comment by eguiraud** [comment]
Uhm, gcc is ok with that code, I'll check whether clang disagrees on my laptop as well.

**Comment by eguiraud** [comment]
Yeah it's a clang vs gcc thing, not sure who's right, not sure what clang is even complaining about exactly :smile: WIP...

**Comment by eguiraud** [comment]
Uh, this compiles with gcc 12 and C++17, does not compile with clang 13 or gcc 12 and C++14:

```cpp
#include <vector>

template <template <typename> class Coll_t> int foo() {
  Coll_t<int> t;
  return t.size();
}

int main() { return foo<std::vector>(); }
```

I have to find another way to parameterize the test case over the collection type.

File: tree/ntuple/v7/test/rfield_vector.cxx [hunk 1]
```diff
@@ -1,51 +1,99 @@
...
    }
 
    RNTupleReader ntuple(std::make_unique<RPageSourceFile>("myNTuple", fileGuard.GetPath(), RNTupleReadOptions()));
-   EXPECT_EQ(1U, ntuple.GetNEntries());
+   EXPECT_EQ(3U, ntuple.GetNEntries());
 
    auto viewKlassVec = ntuple.GetViewCollection("klassVec");
    auto viewKlass = viewKlassVec.GetView<CustomStruct>("_0");
    auto viewKlassA = viewKlassVec.GetView<float>("_0.a");
+   auto viewKlassV1 = viewKlassVec.GetView<std::vector<float>>("_0.v1");
```
Comment (jblomer) [inline tree/ntuple/v7/test/rfield_vector.cxx:42]: We could read one of the next three fields as `RVec` (to check reading `RVec` as `std::vector` and vice versa)

File: tree/ntuple/v7/test/rfield_vector.cxx [hunk 2]
```diff
@@ -1,51 +1,99 @@
...
-TEST(RNTuple, ClassVector)
+// A layer of indirection to hide std::vector's second template parameter.
+// This way we can generalize tests over RVec and std::vector using a template template parameter (see below).
+template <typename T>
+using Vector_t = std::vector<T>;
+
+template <template <typename> class Coll_t>
+void TestClassVector()
 {
    FileRaii fileGuard("test_ntuple_classvector.root");
```
Comment (jblomer) [inline tree/ntuple/v7/test/rfield_vector.cxx:11]: Should use different file names for the different invocations.

**Review by jblomer — CHANGES_REQUESTED** [review]
Very nice!  Some minor comments.

You're right, `RField<RVec>` treats resize correctly, `RField<std::vector>` doesn't. To be fixed.

File: tree/ntuple/v7/test/rfield_vector.cxx (hunk 2 above)
Comment (eguiraud) [reply tree/ntuple/v7/test/rfield_vector.cxx:11]: done!

File: tree/ntuple/v7/test/rfield_vector.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/ntuple/v7/test/rfield_vector.cxx:42]: uhm i'm not sure I understand the suggestion, could you propose a change? (there is another test for reading std::vectors as RVec, I don't know if we have tests for reading RVecs as std::vectors -- should I just change `std::vector<float>` to `ROOT::RVecF` at line 42 to exercise that path?)

**Comment by jblomer** [comment]
Replaced by #12836

**Pull request closed without merging**